sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1769,7 +1769,10 @@ class Dataset[T] private[sql](
    * @since 2.0.0
    */
   def dropDuplicates(colNames: Seq[String]): Dataset[T] = withTypedPlan {
-    val groupCols = colNames.map(resolve)
+    val groupCols = colNames.map {
+      case column if column.contains(".") && !column.contains("`") => s"`$column`"
```

Comment:
BTW, adding backticks and parsing it is not a good way, please at least copy the code from `withColumn` or `drop`.